src/software/ai/navigator/path_manager/path_manager.h

```diff
@@ -0,0 +1,58 @@
+#pragma once
+#include <vector>
+
+#include "software/ai/navigator/obstacle/obstacle.h"
+#include "software/ai/navigator/path_planner/path_planner.h"
+#include "software/geom/point.h"
+#include "software/geom/rectangle.h"
+#include "software/geom/spline.h"
+
+/**
+ * PathManager is an interface for a path manager that,
+ * given a list of start and destination point pairs,
+ * a list of static obstacles, and navigable_area will return
+ * a list of 'best' paths between each of the pairs of
+ * start and destination
+ */
+
+using Path = std::optional<Spline>;
```

Comment:
could we change this to `using Path = Spline`? I think that's more in line with what a user of this type would expect?

Reply:
that's not supposed to be there...

Reply:
Looks like this is also defined on line 16 of `path_planner.h`?

Reply:
Yup

Reply:
Two things:
- should we have that defined in a separate file `Path.h` (I didn't think of this on the original review when we added it)?
- my original comment about the types not being what I would expect still stands.

Additionally, I know replying to all the comments is a pain, but stuff like this really warrants at least a sentence explanation justifying why that's there and why my comment doesn't apply to it.

Reply:
yeah my bad, yeah it's defined in path_planner.h and it's optional because that's how we return no path. Move this typedef and pathobjective to a new file like path.h?

Reply:
moved to path_objective.h

Reply:
- looks like this is in `path_planner.h`? 
- would you object to changing to `using Path = Spline`? I think it's easier to understand / more obvious if a function returns a `std::optional<Path>`, as we generally don't typedef out the `optional`

Reply:
Done